lib/saluki-components/src/sources/checks/builder/python.rs

```diff
@@ -0,0 +1,312 @@
+                                    }
+                                    project_root
+                                }
+                                Err(e) => {
+                                    error!("Failed to get current executable path: {}", e);
+                                    Path::new(".").to_path_buf()
+                                }
+                            };
+
+                            // Add paths relative to executable location
+                            let dist_path = project_root.join("dist");
+                            let checks_d_path = dist_path.join("checks.d");
+
+                            p.insert(0, dist_path.to_string_lossy().as_ref()).unwrap(); // common modules are shipped in the dist path directly or under the "checks/" sub-dir
+                            p.insert(0, checks_d_path.to_string_lossy().as_ref()).unwrap(); // integrations-core legacy checks
+                            p.insert(0, "/etc/datadog-agent/checks.d/").unwrap(); // Agent checks folder
+
+                            info!("Python sys.path is: {:?}", p);
+                        }
+                        Err(e) => {
+                            error!(%e, "Could not get sys.path");
+                            return None;
+                        }
+                    };
+                }
+            } else {
+                error!("Could not import sys module");
+                return None;
+            }
+
+            // Validate that python env is correctly configured
+            let modd = match py.import("datadog_checks.checks") {
+                Ok(m) => m,
+                Err(e) => {
+                    if let Some(traceback) = e.traceback(py) {
+                        error!("Traceback: {}", traceback.format().expect("Could format traceback"));
+                    }
+                    error!(%e, "Could not import datadog_checks module");
+                    return None;
+                }
+            };
+            match modd.getattr("AgentCheck") {
+                Ok(c) => {
+                    info!("All pre-requisites for running python checks.");
+                    Some(c.unbind())
+                }
+                Err(e) => {
+                    if let Some(traceback) = e.traceback(py) {
+                        error!("Traceback: {}", traceback.format().expect("Could format traceback"));
+                    }
+                    error!(%e, "Could not get AgentCheck class.");
+                    None
+                }
+            }
+        });
+
+        match result {
+            Some(agent_check_class) => Self {
+                valid: true,
+                agent_check_class: Some(agent_check_class),
+            },
+            None => Self {
+                valid: false,
+                agent_check_class: None,
+            },
+        }
+    }
+}
+
+#[async_trait]
+impl CheckBuilder for PythonCheckBuilder {
+    async fn build_check(
```

Comment:
Why did we make `CheckBuilder::build_check` async? I don't see us doing anything asynchronous here.

Reply:
It is a leftover from when I was trying to lazy-load the Python interpreter. I can remove it

Reply:
I changed the function not to be async, and made the Python environment check lazy loaded using `OnceLock`
https://github.com/DataDog/saluki/pull/691/commits/df816042672a7d7b9c0a8d4e899ba5bafef74375